dali/pipeline/operators/decoder/host_decoder.h

```diff
@@ -19,55 +19,97 @@
 
 #include "dali/common.h"
 #include "dali/error_handling.h"
-#include "dali/pipeline/operators/operator.h"
 #include "dali/image/jpeg.h"
 #include "dali/image/png.h"
+#include "dali/pipeline/operators/operator.h"
 
 namespace dali {
 
 class HostDecoder : public Operator<CPUBackend> {
  public:
-  explicit inline HostDecoder(const OpSpec &spec) :
-    Operator<CPUBackend>(spec),
-    output_type_(spec.GetArgument<DALIImageType>("output_type")),
-    c_(IsColor(output_type_) ? 3 : 1) {}
+  explicit inline HostDecoder(const OpSpec &spec)
+      : Operator<CPUBackend>(spec),
+        output_type_(spec.GetArgument<DALIImageType>("output_type")),
+        c_(IsColor(output_type_) ? 3 : 1),
+        decode_sequences_(spec.GetArgument<bool>("decode_sequences")) {}
```

Comment:
This Decoder serves now for two purposes: decode images and decode sequences. It might be a good idea to separate these two tasks. In my opinion, this is very intuitive situation for inheritance, e.g. `Decoder` as base abstraction with `ImageDecoder` and `SequenceDecoder` as derived classes. I can easily imagine, that in the nearest future we'd have some other types of data to handle, like audio data, point cloud data, etc. This way we'll have easily modifiable code to handle these things.
On top of that, as @pribalta pointed out in another PR, flag members in classes, that describe objects' state are usually sign of flaws in design.

Reply:
Decoding sequences was incorporated into SequenceDecoder.